sql/tests: TestRandomSyntaxGeneration failed

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8208463?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8208463?buildTab=artifacts#/) on master @ [642afd6e8c8d1f967da4aa0c3c08e2bdc0495100](https://github.com/cockroachdb/cockroach/commits/642afd6e8c8d1f967da4aa0c3c08e2bdc0495100):


```
crypto/tls.(*atLeastReader).Read(0xc05661a8b8, {0xc04a583d40?, 0x0?, 0x389488168619053b?})
	GOROOT/src/crypto/tls/conn.go:787 +0x3d fp=0xc0339d0938 sp=0xc0339d08f0 pc=0x86799d
bytes.(*Buffer).ReadFrom(0xc0569a3078, {0x6c2e660, 0xc05661a8b8})
	GOROOT/src/bytes/buffer.go:202 +0x98 fp=0xc0339d0990 sp=0xc0339d0938 pc=0x565858
crypto/tls.(*Conn).readFromUntil(0xc0569a2e00, {0x6c3dd20?, 0xc04bd20b18}, 0x80deb1fe72be5d74?)
	GOROOT/src/crypto/tls/conn.go:809 +0xe5 fp=0xc0339d09d0 sp=0xc0339d0990 pc=0x867b85
crypto/tls.(*Conn).readRecordOrCCS(0xc0569a2e00, 0x0)
	GOROOT/src/crypto/tls/conn.go:616 +0x116 fp=0xc0339d0d58 sp=0xc0339d09d0 pc=0x864fd6
crypto/tls.(*Conn).readRecord(...)
	GOROOT/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).readHandshake(0xc0569a2e00)
	GOROOT/src/crypto/tls/conn.go:1019 +0x6d fp=0xc0339d0e38 sp=0xc0339d0d58 pc=0x868ded
crypto/tls.(*serverHandshakeStateTLS13).readClientCertificate(0xc0339d11c0)
	GOROOT/src/crypto/tls/handshake_server_tls13.go:787 +0x65 fp=0xc0339d10c0 sp=0xc0339d0e38 pc=0x8948e5
crypto/tls.(*serverHandshakeStateTLS13).handshake(0xc0339d11c0)
	GOROOT/src/crypto/tls/handshake_server_tls13.go:78 +0xb6 fp=0xc0339d10e0 sp=0xc0339d10c0 pc=0x890bb6
crypto/tls.(*Conn).serverHandshake(0xc0569a2e00, {0x6c75730, 0xc0569a0b00})
	GOROOT/src/crypto/tls/handshake_server.go:54 +0xd0 fp=0xc0339d12b8 sp=0xc0339d10e0 pc=0x88b550
crypto/tls.(*Conn).serverHandshake-fm({0x6c75730?, 0xc0569a0b00?})
	<autogenerated>:1 +0x39 fp=0xc0339d12e0 sp=0xc0339d12b8 pc=0x89f4d9
crypto/tls.(*Conn).handshakeContext(0xc0569a2e00, {0x6c75768, 0xc000080048})
	GOROOT/src/crypto/tls/conn.go:1462 +0x32f fp=0xc0339d14a8 sp=0xc0339d12e0 pc=0x86b6cf
crypto/tls.(*Conn).HandshakeContext(...)
	GOROOT/src/crypto/tls/conn.go:1405
crypto/tls.(*Conn).Handshake(...)
	GOROOT/src/crypto/tls/conn.go:1389
crypto/tls.(*Conn).Read(0xc0569a2e00, {0xc0569c4438, 0x4, 0x493f10?})
	GOROOT/src/crypto/tls/conn.go:1274 +0x65 fp=0xc0339d1518 sp=0xc0339d14a8 pc=0x86ab25
io.ReadAtLeast({0x7eff005e27f8, 0xc0569a2e00}, {0xc0569c4438, 0x4, 0x4}, 0x4)
	GOROOT/src/io/io.go:332 +0x9a fp=0xc0339d1560 sp=0xc0339d1518 pc=0x50ccba
io.ReadFull(...)
	GOROOT/src/io/io.go:351
github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirebase.(*ReadBuffer).ReadUntypedMsg(0xc0569c4420, {0x7eff005e27f8, 0xc0569a2e00})
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirebase/encoding.go:155 +0x65 fp=0xc0339d1720 sp=0xc0339d1560 pc=0x27e6c85
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).readVersion(0xc000cf23c0, {0x7eff005e27f8, 0xc0569a2e00})
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1371 +0x127 fp=0xc0339d17a8 sp=0xc0339d1720 pc=0x38ce567
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).maybeUpgradeToSecureConn(0xc000cf23c0, {0x0?, 0xffc?}, {0x6cbb9a8, 0xc04bd20b18}, 0x2, 0x6c3dd20?, 0xc0339d1bc0)
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1315 +0x35d fp=0xc0339d1878 sp=0xc0339d17a8 pc=0x38cdd9d
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn(0xc000cf23c0, {0x6c75730, 0xc056b06880}, {0x6cbb9a8?, 0xc04bd20b18}, 0x6a?)
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:811 +0x59e fp=0xc0339d1e20 sp=0xc0339d1878 pc=0x38ca83e
github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).startServeSQL.func1.1({0x6c757d8?, 0xc0568207b0?}, {0x6cbb9a8, 0xc04bd20b18})
	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1568 +0xc8 fp=0xc0339d1eb0 sp=0xc0339d1e20 pc=0x3b95648
github.com/cockroachdb/cockroach/pkg/util/netutil.(*TCPServer).ServeWith.func1({0x6c757d8, 0xc0568207b0})
	github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:188 +0xc7 fp=0xc0339d1f30 sp=0xc0339d1eb0 pc=0x17d96a7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146 fp=0xc0339d1fe0 sp=0xc0339d1f30 pc=0x10fdae6
runtime.goexit()
	GOROOT/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0339d1fe8 sp=0xc0339d1fe0 pc=0x4c4021
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-sessions
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23617

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8230919?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8230919?buildTab=artifacts#/) on master @ [9b567e3439e49c109e5204c9d0afec49c1a3169b](https://github.com/cockroachdb/cockroach/commits/9b567e3439e49c109e5204c9d0afec49c1a3169b):


```
    rsg_test.go:822: 1m15s of 5m0s: 205880 executions, 11490 successful
    rsg_test.go:822: 1m20s of 5m0s: 219087 executions, 12174 successful
    rsg_test.go:822: 1m25s of 5m0s: 232805 executions, 12884 successful
    rsg_test.go:822: 1m30s of 5m0s: 247093 executions, 13582 successful
    rsg_test.go:822: 1m35s of 5m0s: 260932 executions, 14262 successful
    rsg_test.go:822: 1m40s of 5m0s: 274549 executions, 14923 successful
    rsg_test.go:822: 1m45s of 5m0s: 287695 executions, 15500 successful
    rsg_test.go:822: 1m50s of 5m0s: 300602 executions, 16089 successful
    rsg_test.go:822: 1m55s of 5m0s: 314228 executions, 16669 successful
    rsg_test.go:822: 2m0s of 5m0s: 327497 executions, 17223 successful
    rsg_test.go:822: 2m5s of 5m0s: 340470 executions, 17793 successful
    rsg_test.go:822: 2m10s of 5m0s: 354193 executions, 18425 successful
    rsg_test.go:822: 2m15s of 5m0s: 368429 executions, 19085 successful
    rsg_test.go:822: 2m20s of 5m0s: 383271 executions, 19773 successful
    rsg_test.go:822: 2m25s of 5m0s: 397220 executions, 20425 successful
    rsg_test.go:822: 2m30s of 5m0s: 410873 executions, 21082 successful
    rsg_test.go:822: 2m35s of 5m0s: 424312 executions, 21739 successful
    rsg_test.go:822: 2m40s of 5m0s: 438255 executions, 22432 successful
    rsg_test.go:822: 2m45s of 5m0s: 452451 executions, 23104 successful
    rsg_test.go:822: 2m50s of 5m0s: 466039 executions, 23673 successful
    rsg_test.go:822: 2m55s of 5m0s: 479611 executions, 24266 successful
    rsg_test.go:822: 3m0s of 5m0s: 493385 executions, 24848 successful
    rsg_test.go:822: 3m5s of 5m0s: 506921 executions, 25424 successful
    rsg_test.go:822: 3m10s of 5m0s: 520687 executions, 25978 successful
    rsg_test.go:822: 3m15s of 5m0s: 533585 executions, 26466 successful
    rsg_test.go:822: 3m20s of 5m0s: 546786 executions, 26998 successful
    rsg_test.go:822: 3m25s of 5m0s: 559353 executions, 27491 successful
    rsg_test.go:822: 3m30s of 5m0s: 571622 executions, 28022 successful
    rsg_test.go:822: 3m35s of 5m0s: 584567 executions, 28506 successful
    rsg_test.go:822: 3m40s of 5m0s: 597316 executions, 28957 successful
    rsg_test.go:822: 3m45s of 5m0s: 610085 executions, 29466 successful
    rsg_test.go:822: 3m50s of 5m0s: 622064 executions, 30002 successful
    rsg_test.go:822: 3m55s of 5m0s: 633868 executions, 30480 successful
    rsg_test.go:822: 4m0s of 5m0s: 646130 executions, 30978 successful
    rsg_test.go:822: 4m5s of 5m0s: 657640 executions, 31460 successful
    rsg_test.go:822: 4m10s of 5m0s: 669414 executions, 31954 successful
    rsg_test.go:822: 4m15s of 5m0s: 681412 executions, 32470 successful
    rsg_test.go:822: 4m20s of 5m0s: 693069 executions, 33003 successful
    rsg_test.go:822: 4m25s of 5m0s: 704699 executions, 33485 successful
    rsg_test.go:822: 4m30s of 5m0s: 718687 executions, 34181 successful
    rsg_test.go:822: 4m35s of 5m0s: 732227 executions, 34799 successful
    rsg_test.go:822: 4m40s of 5m0s: 745839 executions, 35392 successful
    rsg_test.go:822: 4m45s of 5m0s: 759042 executions, 35954 successful
    rsg_test.go:822: 4m50s of 5m0s: 771743 executions, 36489 successful
    rsg_test.go:822: 4m55s of 5m0s: 783832 executions, 37095 successful
    rsg_test.go:858: 796746 executions, 37770 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP TABLE * INTO 'string' IN PLACEHOLDER WITH OPTIONS ( DETACHED = FALSE )", formattedSQL="BACKUP TABLE * INTO 'string' IN 'placeholder' WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration2894180145
--- FAIL: TestRandomSyntaxGeneration (301.45s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8223831?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8223831?buildTab=artifacts#/) on master @ [7594fb2d1438dd5ed8f80a887b8d60b6cdd296ed](https://github.com/cockroachdb/cockroach/commits/7594fb2d1438dd5ed8f80a887b8d60b6cdd296ed):


```
    rsg_test.go:822: 1m20s of 5m0s: 209356 executions, 11180 successful
    rsg_test.go:822: 1m25s of 5m0s: 223352 executions, 11810 successful
    rsg_test.go:822: 1m30s of 5m0s: 237775 executions, 12388 successful
    rsg_test.go:822: 1m35s of 5m0s: 251610 executions, 12982 successful
    rsg_test.go:822: 1m40s of 5m0s: 264483 executions, 13512 successful
    rsg_test.go:822: 1m45s of 5m0s: 277604 executions, 14085 successful
    rsg_test.go:822: 1m50s of 5m0s: 290289 executions, 14701 successful
    rsg_test.go:822: 1m55s of 5m0s: 303025 executions, 15336 successful
    rsg_test.go:822: 2m0s of 5m0s: 315919 executions, 16001 successful
    rsg_test.go:822: 2m5s of 5m0s: 329783 executions, 16699 successful
    rsg_test.go:822: 2m10s of 5m0s: 343492 executions, 17368 successful
    rsg_test.go:822: 2m15s of 5m0s: 357517 executions, 18038 successful
    rsg_test.go:822: 2m20s of 5m0s: 370895 executions, 18693 successful
    rsg_test.go:822: 2m25s of 5m0s: 385007 executions, 19365 successful
    rsg_test.go:822: 2m30s of 5m0s: 398762 executions, 20004 successful
    rsg_test.go:822: 2m35s of 5m0s: 412050 executions, 20589 successful
    rsg_test.go:822: 2m40s of 5m0s: 424976 executions, 21173 successful
    rsg_test.go:822: 2m45s of 5m0s: 437414 executions, 21733 successful
    rsg_test.go:822: 2m50s of 5m0s: 450357 executions, 22277 successful
    rsg_test.go:822: 2m55s of 5m0s: 463357 executions, 22823 successful
    rsg_test.go:822: 3m0s of 5m0s: 476160 executions, 23445 successful
    rsg_test.go:822: 3m5s of 5m0s: 489098 executions, 24074 successful
    rsg_test.go:822: 3m10s of 5m0s: 502630 executions, 24712 successful
    rsg_test.go:822: 3m15s of 5m0s: 516643 executions, 25407 successful
    rsg_test.go:822: 3m20s of 5m0s: 530497 executions, 26052 successful
    rsg_test.go:822: 3m25s of 5m0s: 544401 executions, 26761 successful
    rsg_test.go:822: 3m30s of 5m0s: 557982 executions, 27371 successful
    rsg_test.go:822: 3m35s of 5m0s: 571947 executions, 27949 successful
    rsg_test.go:822: 3m40s of 5m0s: 585304 executions, 28505 successful
    rsg_test.go:822: 3m45s of 5m0s: 598832 executions, 29090 successful
    rsg_test.go:822: 3m50s of 5m0s: 612041 executions, 29617 successful
    rsg_test.go:822: 3m55s of 5m0s: 624270 executions, 30127 successful
    rsg_test.go:822: 4m0s of 5m0s: 637466 executions, 30625 successful
    rsg_test.go:822: 4m5s of 5m0s: 650770 executions, 31185 successful
    rsg_test.go:822: 4m10s of 5m0s: 664413 executions, 31718 successful
    rsg_test.go:822: 4m15s of 5m0s: 677711 executions, 32258 successful
    rsg_test.go:822: 4m20s of 5m0s: 690856 executions, 32797 successful
    rsg_test.go:822: 4m25s of 5m0s: 704005 executions, 33331 successful
    rsg_test.go:822: 4m30s of 5m0s: 717125 executions, 33860 successful
    rsg_test.go:822: 4m35s of 5m0s: 730476 executions, 34438 successful
    rsg_test.go:822: 4m40s of 5m0s: 743900 executions, 34990 successful
    rsg_test.go:822: 4m45s of 5m0s: 757001 executions, 35548 successful
    rsg_test.go:822: 4m50s of 5m0s: 769643 executions, 36065 successful
    rsg_test.go:822: 4m55s of 5m0s: 782449 executions, 36593 successful
    rsg_test.go:822: 5m0s of 5m0s: 794685 executions, 37069 successful
    rsg_test.go:858: 794686 executions, 37069 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP ISERROR TO PLACEHOLDER INCREMENTAL FROM UPSERT , SEQUENCES WITH OPTIONS ( DETACHED = FALSE )", formattedSQL="BACKUP TABLE \"iserror\" TO 'placeholder' INCREMENTAL FROM 'upsert', 'sequences' WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1787880193
--- FAIL: TestRandomSyntaxGeneration (301.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8255177?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8255177?buildTab=artifacts#/) on master @ [b5cffc06bf36ce42b601d3307a1e38f933be40c4](https://github.com/cockroachdb/cockroach/commits/b5cffc06bf36ce42b601d3307a1e38f933be40c4):


```
    rsg_test.go:822: 1m20s of 5m0s: 205716 executions, 11024 successful
    rsg_test.go:822: 1m25s of 5m0s: 218897 executions, 11680 successful
    rsg_test.go:822: 1m30s of 5m0s: 232691 executions, 12389 successful
    rsg_test.go:822: 1m35s of 5m0s: 246930 executions, 13078 successful
    rsg_test.go:822: 1m40s of 5m0s: 260832 executions, 13772 successful
    rsg_test.go:822: 1m45s of 5m0s: 274665 executions, 14426 successful
    rsg_test.go:822: 1m50s of 5m0s: 288391 executions, 15083 successful
    rsg_test.go:822: 1m55s of 5m0s: 302415 executions, 15746 successful
    rsg_test.go:822: 2m0s of 5m0s: 316901 executions, 16435 successful
    rsg_test.go:822: 2m5s of 5m0s: 331097 executions, 17188 successful
    rsg_test.go:822: 2m10s of 5m0s: 344725 executions, 17879 successful
    rsg_test.go:822: 2m15s of 5m0s: 358891 executions, 18539 successful
    rsg_test.go:822: 2m20s of 5m0s: 373359 executions, 19248 successful
    rsg_test.go:822: 2m25s of 5m0s: 387912 executions, 19929 successful
    rsg_test.go:822: 2m30s of 5m0s: 401450 executions, 20583 successful
    rsg_test.go:822: 2m35s of 5m0s: 415785 executions, 21216 successful
    rsg_test.go:822: 2m40s of 5m0s: 429654 executions, 21829 successful
    rsg_test.go:822: 2m45s of 5m0s: 443571 executions, 22438 successful
    rsg_test.go:822: 2m50s of 5m0s: 457095 executions, 23028 successful
    rsg_test.go:822: 2m55s of 5m0s: 470518 executions, 23708 successful
    rsg_test.go:822: 3m0s of 5m0s: 483514 executions, 24258 successful
    rsg_test.go:822: 3m5s of 5m0s: 496782 executions, 24801 successful
    rsg_test.go:822: 3m10s of 5m0s: 509355 executions, 25338 successful
    rsg_test.go:822: 3m15s of 5m0s: 523331 executions, 25954 successful
    rsg_test.go:822: 3m20s of 5m0s: 538042 executions, 26675 successful
    rsg_test.go:822: 3m25s of 5m0s: 552126 executions, 27301 successful
    rsg_test.go:822: 3m30s of 5m0s: 566246 executions, 27915 successful
    rsg_test.go:822: 3m35s of 5m0s: 580265 executions, 28591 successful
    rsg_test.go:822: 3m40s of 5m0s: 593891 executions, 29208 successful
    rsg_test.go:822: 3m45s of 5m0s: 607388 executions, 29869 successful
    rsg_test.go:822: 3m50s of 5m0s: 621551 executions, 30559 successful
    rsg_test.go:822: 3m55s of 5m0s: 635433 executions, 31162 successful
    rsg_test.go:822: 4m0s of 5m0s: 649105 executions, 31801 successful
    rsg_test.go:822: 4m5s of 5m0s: 662825 executions, 32441 successful
    rsg_test.go:822: 4m10s of 5m0s: 676930 executions, 33140 successful
    rsg_test.go:822: 4m15s of 5m0s: 690492 executions, 33832 successful
    rsg_test.go:822: 4m20s of 5m0s: 703737 executions, 34456 successful
    rsg_test.go:822: 4m25s of 5m0s: 717067 executions, 35147 successful
    rsg_test.go:822: 4m30s of 5m0s: 731111 executions, 35813 successful
    rsg_test.go:822: 4m35s of 5m0s: 744762 executions, 36439 successful
    rsg_test.go:822: 4m40s of 5m0s: 758657 executions, 37130 successful
    rsg_test.go:822: 4m45s of 5m0s: 772847 executions, 37751 successful
    rsg_test.go:822: 4m50s of 5m0s: 786869 executions, 38401 successful
    rsg_test.go:822: 4m55s of 5m0s: 801409 executions, 38987 successful
    rsg_test.go:822: 5m0s of 5m0s: 815621 executions, 39576 successful
    rsg_test.go:858: 815623 executions, 39576 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP INTO 'string' IN ( PLACEHOLDER ) WITH DETACHED = FALSE", formattedSQL="BACKUP INTO 'string' IN 'placeholder' WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration3791194350
--- FAIL: TestRandomSyntaxGeneration (301.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8240120?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8240120?buildTab=artifacts#/) on master @ [27ebfa830ac7122fc3b286136158fef615b20b8f](https://github.com/cockroachdb/cockroach/commits/27ebfa830ac7122fc3b286136158fef615b20b8f):


```
    rsg_test.go:822: 1m20s of 5m0s: 220951 executions, 11891 successful
    rsg_test.go:822: 1m25s of 5m0s: 235145 executions, 12604 successful
    rsg_test.go:822: 1m30s of 5m0s: 249566 executions, 13298 successful
    rsg_test.go:822: 1m35s of 5m0s: 264170 executions, 14032 successful
    rsg_test.go:822: 1m40s of 5m0s: 279252 executions, 14742 successful
    rsg_test.go:822: 1m45s of 5m0s: 293634 executions, 15479 successful
    rsg_test.go:822: 1m50s of 5m0s: 307361 executions, 16176 successful
    rsg_test.go:822: 1m55s of 5m0s: 321664 executions, 16862 successful
    rsg_test.go:822: 2m0s of 5m0s: 335920 executions, 17585 successful
    rsg_test.go:822: 2m5s of 5m0s: 350330 executions, 18241 successful
    rsg_test.go:822: 2m10s of 5m0s: 364422 executions, 18941 successful
    rsg_test.go:822: 2m15s of 5m0s: 378942 executions, 19617 successful
    rsg_test.go:822: 2m20s of 5m0s: 393096 executions, 20266 successful
    rsg_test.go:822: 2m25s of 5m0s: 406960 executions, 20809 successful
    rsg_test.go:822: 2m30s of 5m0s: 420215 executions, 21325 successful
    rsg_test.go:822: 2m35s of 5m0s: 432969 executions, 21878 successful
    rsg_test.go:822: 2m40s of 5m0s: 445914 executions, 22417 successful
    rsg_test.go:822: 2m45s of 5m0s: 459010 executions, 22936 successful
    rsg_test.go:822: 2m50s of 5m0s: 472365 executions, 23486 successful
    rsg_test.go:822: 2m55s of 5m0s: 485187 executions, 24044 successful
    rsg_test.go:822: 3m0s of 5m0s: 499864 executions, 24719 successful
    rsg_test.go:822: 3m5s of 5m0s: 513856 executions, 25381 successful
    rsg_test.go:822: 3m10s of 5m0s: 528119 executions, 26060 successful
    rsg_test.go:822: 3m15s of 5m0s: 542622 executions, 26743 successful
    rsg_test.go:822: 3m20s of 5m0s: 557545 executions, 27408 successful
    rsg_test.go:822: 3m25s of 5m0s: 571998 executions, 28096 successful
    rsg_test.go:822: 3m30s of 5m0s: 586608 executions, 28764 successful
    rsg_test.go:822: 3m35s of 5m0s: 600153 executions, 29350 successful
    rsg_test.go:822: 3m40s of 5m0s: 614209 executions, 30020 successful
    rsg_test.go:822: 3m45s of 5m0s: 628239 executions, 30608 successful
    rsg_test.go:822: 3m50s of 5m0s: 641915 executions, 31185 successful
    rsg_test.go:822: 3m55s of 5m0s: 655864 executions, 31780 successful
    rsg_test.go:822: 4m0s of 5m0s: 669694 executions, 32346 successful
    rsg_test.go:822: 4m5s of 5m0s: 683408 executions, 32925 successful
    rsg_test.go:822: 4m10s of 5m0s: 697724 executions, 33483 successful
    rsg_test.go:822: 4m15s of 5m0s: 711784 executions, 34097 successful
    rsg_test.go:822: 4m20s of 5m0s: 725986 executions, 34692 successful
    rsg_test.go:822: 4m25s of 5m0s: 740174 executions, 35272 successful
    rsg_test.go:822: 4m30s of 5m0s: 754217 executions, 35820 successful
    rsg_test.go:822: 4m35s of 5m0s: 768123 executions, 36313 successful
    rsg_test.go:822: 4m40s of 5m0s: 782605 executions, 36869 successful
    rsg_test.go:822: 4m45s of 5m0s: 796585 executions, 37447 successful
    rsg_test.go:822: 4m50s of 5m0s: 809865 executions, 37992 successful
    rsg_test.go:822: 4m55s of 5m0s: 823277 executions, 38551 successful
    rsg_test.go:822: 5m0s of 5m0s: 835924 executions, 39089 successful
    rsg_test.go:858: 835925 executions, 39089 successful
    rsg_test.go:866: cannot parse output of Format: sql="ALTER TENANT INTERVAL 'string' MINUTE RESET CLUSTER SETTING ident . POINT . RIGHT . IFNULL . IMMEDIATE . AUTHORIZATION", formattedSQL="ALTER TENANT 'string'::INTERVAL MINUTE SET CLUSTER SETTING \"ident.point.right.ifnull.immediate.authorization\" = DEFAULT": at or near ":": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1486489256
--- FAIL: TestRandomSyntaxGeneration (301.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8270431?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8270431?buildTab=artifacts#/) on master @ [cec8af84bf01918d28b1e06a29b18ac9bee7cbb8](https://github.com/cockroachdb/cockroach/commits/cec8af84bf01918d28b1e06a29b18ac9bee7cbb8):


```
    rsg_test.go:822: 1m15s of 5m0s: 222194 executions, 12138 successful
    rsg_test.go:822: 1m20s of 5m0s: 237281 executions, 12868 successful
    rsg_test.go:822: 1m25s of 5m0s: 253162 executions, 13621 successful
    rsg_test.go:822: 1m30s of 5m0s: 268237 executions, 14368 successful
    rsg_test.go:822: 1m35s of 5m0s: 283807 executions, 15083 successful
    rsg_test.go:822: 1m40s of 5m0s: 299295 executions, 15701 successful
    rsg_test.go:822: 1m45s of 5m0s: 314847 executions, 16384 successful
    rsg_test.go:822: 1m50s of 5m0s: 330107 executions, 17132 successful
    rsg_test.go:822: 1m55s of 5m0s: 345633 executions, 17916 successful
    rsg_test.go:822: 2m0s of 5m0s: 361000 executions, 18656 successful
    rsg_test.go:822: 2m5s of 5m0s: 376308 executions, 19397 successful
    rsg_test.go:822: 2m10s of 5m0s: 391747 executions, 20095 successful
    rsg_test.go:822: 2m15s of 5m0s: 407425 executions, 20900 successful
    rsg_test.go:822: 2m20s of 5m0s: 423099 executions, 21645 successful
    rsg_test.go:822: 2m25s of 5m0s: 437883 executions, 22272 successful
    rsg_test.go:822: 2m30s of 5m0s: 453098 executions, 23056 successful
    rsg_test.go:822: 2m35s of 5m0s: 468917 executions, 23781 successful
    rsg_test.go:822: 2m40s of 5m0s: 484680 executions, 24477 successful
    rsg_test.go:822: 2m45s of 5m0s: 499948 executions, 25114 successful
    rsg_test.go:822: 2m50s of 5m0s: 515651 executions, 25841 successful
    rsg_test.go:822: 2m55s of 5m0s: 530706 executions, 26574 successful
    rsg_test.go:822: 3m0s of 5m0s: 545836 executions, 27299 successful
    rsg_test.go:822: 3m5s of 5m0s: 561172 executions, 27977 successful
    rsg_test.go:822: 3m10s of 5m0s: 576240 executions, 28691 successful
    rsg_test.go:822: 3m15s of 5m0s: 592066 executions, 29431 successful
    rsg_test.go:822: 3m20s of 5m0s: 607643 executions, 30139 successful
    rsg_test.go:822: 3m25s of 5m0s: 623117 executions, 30895 successful
    rsg_test.go:822: 3m30s of 5m0s: 638582 executions, 31588 successful
    rsg_test.go:822: 3m35s of 5m0s: 654336 executions, 32308 successful
    rsg_test.go:822: 3m40s of 5m0s: 669980 executions, 33084 successful
    rsg_test.go:822: 3m45s of 5m0s: 685360 executions, 33790 successful
    rsg_test.go:822: 3m50s of 5m0s: 700672 executions, 34488 successful
    rsg_test.go:822: 3m55s of 5m0s: 715967 executions, 35245 successful
    rsg_test.go:822: 4m0s of 5m0s: 731588 executions, 35943 successful
    rsg_test.go:822: 4m5s of 5m0s: 746568 executions, 36645 successful
    rsg_test.go:822: 4m10s of 5m0s: 760698 executions, 37300 successful
    rsg_test.go:822: 4m15s of 5m0s: 775782 executions, 37975 successful
    rsg_test.go:822: 4m20s of 5m0s: 790935 executions, 38621 successful
    rsg_test.go:822: 4m25s of 5m0s: 806508 executions, 39305 successful
    rsg_test.go:822: 4m30s of 5m0s: 822021 executions, 40071 successful
    rsg_test.go:822: 4m35s of 5m0s: 837284 executions, 40811 successful
    rsg_test.go:822: 4m40s of 5m0s: 852076 executions, 41535 successful
    rsg_test.go:822: 4m45s of 5m0s: 866635 executions, 42215 successful
    rsg_test.go:822: 4m50s of 5m0s: 881682 executions, 42881 successful
    rsg_test.go:822: 4m55s of 5m0s: 897114 executions, 43513 successful
    rsg_test.go:858: 911545 executions, 44058 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP ident INTO ( PLACEHOLDER , PLACEHOLDER ) WITH DETACHED = FALSE", formattedSQL="BACKUP TABLE ident INTO ('placeholder', 'placeholder') WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1365964288
--- FAIL: TestRandomSyntaxGeneration (301.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8318160?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8318160?buildTab=artifacts#/) on master @ [8866ecadd99e5f6dc8db9fed5d09bc57cd90b03c](https://github.com/cockroachdb/cockroach/commits/8866ecadd99e5f6dc8db9fed5d09bc57cd90b03c):


```
    rsg_test.go:822: 1m15s of 5m0s: 213385 executions, 11700 successful
    rsg_test.go:822: 1m20s of 5m0s: 227565 executions, 12378 successful
    rsg_test.go:822: 1m25s of 5m0s: 243137 executions, 13134 successful
    rsg_test.go:822: 1m30s of 5m0s: 259141 executions, 13920 successful
    rsg_test.go:822: 1m35s of 5m0s: 274945 executions, 14681 successful
    rsg_test.go:822: 1m40s of 5m0s: 290731 executions, 15409 successful
    rsg_test.go:822: 1m45s of 5m0s: 306242 executions, 16125 successful
    rsg_test.go:822: 1m50s of 5m0s: 321121 executions, 16838 successful
    rsg_test.go:822: 1m55s of 5m0s: 335987 executions, 17519 successful
    rsg_test.go:822: 2m0s of 5m0s: 351760 executions, 18295 successful
    rsg_test.go:822: 2m5s of 5m0s: 367263 executions, 18998 successful
    rsg_test.go:822: 2m10s of 5m0s: 382761 executions, 19762 successful
    rsg_test.go:822: 2m15s of 5m0s: 397941 executions, 20434 successful
    rsg_test.go:822: 2m20s of 5m0s: 412674 executions, 21137 successful
    rsg_test.go:822: 2m25s of 5m0s: 427489 executions, 21786 successful
    rsg_test.go:822: 2m30s of 5m0s: 442504 executions, 22450 successful
    rsg_test.go:822: 2m35s of 5m0s: 457425 executions, 23161 successful
    rsg_test.go:822: 2m40s of 5m0s: 472271 executions, 23880 successful
    rsg_test.go:822: 2m45s of 5m0s: 486674 executions, 24572 successful
    rsg_test.go:822: 2m50s of 5m0s: 502008 executions, 25285 successful
    rsg_test.go:822: 2m55s of 5m0s: 516904 executions, 26029 successful
    rsg_test.go:822: 3m0s of 5m0s: 532271 executions, 26751 successful
    rsg_test.go:822: 3m5s of 5m0s: 547409 executions, 27455 successful
    rsg_test.go:822: 3m10s of 5m0s: 562988 executions, 28133 successful
    rsg_test.go:822: 3m15s of 5m0s: 578982 executions, 28906 successful
    rsg_test.go:822: 3m20s of 5m0s: 595212 executions, 29568 successful
    rsg_test.go:822: 3m25s of 5m0s: 611202 executions, 30321 successful
    rsg_test.go:822: 3m30s of 5m0s: 626133 executions, 30984 successful
    rsg_test.go:822: 3m35s of 5m0s: 641660 executions, 31807 successful
    rsg_test.go:822: 3m40s of 5m0s: 656616 executions, 32519 successful
    rsg_test.go:822: 3m45s of 5m0s: 671917 executions, 33211 successful
    rsg_test.go:822: 3m50s of 5m0s: 687215 executions, 33874 successful
    rsg_test.go:822: 3m55s of 5m0s: 702182 executions, 34566 successful
    rsg_test.go:822: 4m0s of 5m0s: 717417 executions, 35254 successful
    rsg_test.go:822: 4m5s of 5m0s: 733851 executions, 36018 successful
    rsg_test.go:822: 4m10s of 5m0s: 749696 executions, 36773 successful
    rsg_test.go:822: 4m15s of 5m0s: 766069 executions, 37519 successful
    rsg_test.go:822: 4m20s of 5m0s: 782574 executions, 38264 successful
    rsg_test.go:822: 4m25s of 5m0s: 797632 executions, 38933 successful
    rsg_test.go:822: 4m30s of 5m0s: 813816 executions, 39680 successful
    rsg_test.go:822: 4m35s of 5m0s: 830160 executions, 40437 successful
    rsg_test.go:822: 4m40s of 5m0s: 845905 executions, 41083 successful
    rsg_test.go:822: 4m45s of 5m0s: 861529 executions, 41782 successful
    rsg_test.go:822: 4m50s of 5m0s: 877786 executions, 42495 successful
    rsg_test.go:822: 4m55s of 5m0s: 893742 executions, 43176 successful
    rsg_test.go:858: 909671 executions, 43907 successful
    rsg_test.go:866: cannot parse output of Format: sql="SHOW TENANT INTERVAL 'string'", formattedSQL="SHOW TENANT 'string'::INTERVAL": at or near ":": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration605245569
--- FAIL: TestRandomSyntaxGeneration (300.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8286072?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8286072?buildTab=artifacts#/) on master @ [73485e56768d61b14a5aa9b3f880613d7820edb9](https://github.com/cockroachdb/cockroach/commits/73485e56768d61b14a5aa9b3f880613d7820edb9):


```
    rsg_test.go:822: 1m15s of 5m0s: 208522 executions, 10335 successful
    rsg_test.go:822: 1m20s of 5m0s: 223670 executions, 11085 successful
    rsg_test.go:822: 1m25s of 5m0s: 238209 executions, 11825 successful
    rsg_test.go:822: 1m30s of 5m0s: 252795 executions, 12538 successful
    rsg_test.go:822: 1m35s of 5m0s: 267639 executions, 13284 successful
    rsg_test.go:822: 1m40s of 5m0s: 282427 executions, 13948 successful
    rsg_test.go:822: 1m45s of 5m0s: 298498 executions, 14689 successful
    rsg_test.go:822: 1m50s of 5m0s: 313918 executions, 15374 successful
    rsg_test.go:822: 1m55s of 5m0s: 328743 executions, 16055 successful
    rsg_test.go:822: 2m0s of 5m0s: 343369 executions, 16764 successful
    rsg_test.go:822: 2m5s of 5m0s: 358542 executions, 17492 successful
    rsg_test.go:822: 2m10s of 5m0s: 373379 executions, 18163 successful
    rsg_test.go:822: 2m15s of 5m0s: 389007 executions, 18893 successful
    rsg_test.go:822: 2m20s of 5m0s: 404104 executions, 19585 successful
    rsg_test.go:822: 2m25s of 5m0s: 419656 executions, 20325 successful
    rsg_test.go:822: 2m30s of 5m0s: 434792 executions, 21066 successful
    rsg_test.go:822: 2m35s of 5m0s: 449828 executions, 21753 successful
    rsg_test.go:822: 2m40s of 5m0s: 465050 executions, 22465 successful
    rsg_test.go:822: 2m45s of 5m0s: 480158 executions, 23124 successful
    rsg_test.go:822: 2m50s of 5m0s: 496043 executions, 23823 successful
    rsg_test.go:822: 2m55s of 5m0s: 511520 executions, 24551 successful
    rsg_test.go:822: 3m0s of 5m0s: 526848 executions, 25283 successful
    rsg_test.go:822: 3m5s of 5m0s: 542770 executions, 26011 successful
    rsg_test.go:822: 3m10s of 5m0s: 558726 executions, 26728 successful
    rsg_test.go:822: 3m15s of 5m0s: 574031 executions, 27475 successful
    rsg_test.go:822: 3m20s of 5m0s: 589789 executions, 28162 successful
    rsg_test.go:822: 3m25s of 5m0s: 604040 executions, 28785 successful
    rsg_test.go:822: 3m30s of 5m0s: 618417 executions, 29367 successful
    rsg_test.go:822: 3m35s of 5m0s: 632814 executions, 30036 successful
    rsg_test.go:822: 3m40s of 5m0s: 647516 executions, 30689 successful
    rsg_test.go:822: 3m45s of 5m0s: 662222 executions, 31320 successful
    rsg_test.go:822: 3m50s of 5m0s: 677059 executions, 32037 successful
    rsg_test.go:822: 3m55s of 5m0s: 691457 executions, 32695 successful
    rsg_test.go:822: 4m0s of 5m0s: 706254 executions, 33404 successful
    rsg_test.go:822: 4m5s of 5m0s: 721009 executions, 34169 successful
    rsg_test.go:822: 4m10s of 5m0s: 735951 executions, 34873 successful
    rsg_test.go:822: 4m15s of 5m0s: 751362 executions, 35580 successful
    rsg_test.go:822: 4m20s of 5m0s: 767109 executions, 36318 successful
    rsg_test.go:822: 4m25s of 5m0s: 782401 executions, 36983 successful
    rsg_test.go:822: 4m30s of 5m0s: 798020 executions, 37680 successful
    rsg_test.go:822: 4m35s of 5m0s: 813479 executions, 38399 successful
    rsg_test.go:822: 4m40s of 5m0s: 829265 executions, 39127 successful
    rsg_test.go:822: 4m45s of 5m0s: 844252 executions, 39847 successful
    rsg_test.go:822: 4m50s of 5m0s: 858761 executions, 40536 successful
    rsg_test.go:822: 4m55s of 5m0s: 873455 executions, 41206 successful
    rsg_test.go:858: 887625 executions, 41881 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP INTO LATEST IN UNLOGGED WITH OPTIONS ( DETACHED = FALSE )", formattedSQL="BACKUP INTO LATEST IN 'unlogged' WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration4137844509
--- FAIL: TestRandomSyntaxGeneration (301.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8311522?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8311522?buildTab=artifacts#/) on master @ [8866ecadd99e5f6dc8db9fed5d09bc57cd90b03c](https://github.com/cockroachdb/cockroach/commits/8866ecadd99e5f6dc8db9fed5d09bc57cd90b03c):


```
    rsg_test.go:822: 1m15s of 5m0s: 190498 executions, 10658 successful
    rsg_test.go:822: 1m20s of 5m0s: 203909 executions, 11318 successful
    rsg_test.go:822: 1m25s of 5m0s: 217000 executions, 11948 successful
    rsg_test.go:822: 1m30s of 5m0s: 229936 executions, 12643 successful
    rsg_test.go:822: 1m35s of 5m0s: 242840 executions, 13330 successful
    rsg_test.go:822: 1m40s of 5m0s: 256296 executions, 13989 successful
    rsg_test.go:822: 1m45s of 5m0s: 270230 executions, 14677 successful
    rsg_test.go:822: 1m50s of 5m0s: 284078 executions, 15397 successful
    rsg_test.go:822: 1m55s of 5m0s: 297167 executions, 16024 successful
    rsg_test.go:822: 2m0s of 5m0s: 310953 executions, 16721 successful
    rsg_test.go:822: 2m5s of 5m0s: 324803 executions, 17411 successful
    rsg_test.go:822: 2m10s of 5m0s: 338214 executions, 18083 successful
    rsg_test.go:822: 2m15s of 5m0s: 352367 executions, 18791 successful
    rsg_test.go:822: 2m20s of 5m0s: 366023 executions, 19418 successful
    rsg_test.go:822: 2m25s of 5m0s: 379612 executions, 20058 successful
    rsg_test.go:822: 2m30s of 5m0s: 393222 executions, 20682 successful
    rsg_test.go:822: 2m35s of 5m0s: 407079 executions, 21356 successful
    rsg_test.go:822: 2m40s of 5m0s: 420559 executions, 21975 successful
    rsg_test.go:822: 2m45s of 5m0s: 434197 executions, 22612 successful
    rsg_test.go:822: 2m50s of 5m0s: 447870 executions, 23241 successful
    rsg_test.go:822: 2m55s of 5m0s: 461401 executions, 23824 successful
    rsg_test.go:822: 3m0s of 5m0s: 474442 executions, 24424 successful
    rsg_test.go:822: 3m5s of 5m0s: 487578 executions, 25028 successful
    rsg_test.go:822: 3m10s of 5m0s: 500792 executions, 25648 successful
    rsg_test.go:822: 3m15s of 5m0s: 514277 executions, 26241 successful
    rsg_test.go:822: 3m20s of 5m0s: 527135 executions, 26867 successful
    rsg_test.go:822: 3m25s of 5m0s: 540560 executions, 27542 successful
    rsg_test.go:822: 3m30s of 5m0s: 553552 executions, 28147 successful
    rsg_test.go:822: 3m35s of 5m0s: 567143 executions, 28791 successful
    rsg_test.go:822: 3m40s of 5m0s: 581099 executions, 29458 successful
    rsg_test.go:822: 3m45s of 5m0s: 594715 executions, 30100 successful
    rsg_test.go:822: 3m50s of 5m0s: 607992 executions, 30709 successful
    rsg_test.go:822: 3m55s of 5m0s: 620877 executions, 31330 successful
    rsg_test.go:822: 4m0s of 5m0s: 633358 executions, 31934 successful
    rsg_test.go:822: 4m5s of 5m0s: 647358 executions, 32621 successful
    rsg_test.go:822: 4m10s of 5m0s: 661334 executions, 33209 successful
    rsg_test.go:822: 4m15s of 5m0s: 674705 executions, 33860 successful
    rsg_test.go:822: 4m20s of 5m0s: 688019 executions, 34503 successful
    rsg_test.go:822: 4m25s of 5m0s: 700222 executions, 35096 successful
    rsg_test.go:822: 4m30s of 5m0s: 712468 executions, 35692 successful
    rsg_test.go:822: 4m35s of 5m0s: 725844 executions, 36343 successful
    rsg_test.go:822: 4m40s of 5m0s: 739427 executions, 36997 successful
    rsg_test.go:822: 4m45s of 5m0s: 753035 executions, 37667 successful
    rsg_test.go:822: 4m50s of 5m0s: 765689 executions, 38273 successful
    rsg_test.go:822: 4m55s of 5m0s: 779320 executions, 38925 successful
    rsg_test.go:858: 793069 executions, 39573 successful
    rsg_test.go:866: cannot parse output of Format: sql="ALTER TENANT INTERVAL 'string' SET CLUSTER SETTING POSITION . WHERE . INNER . ident TO INDEX_AFTER_ORDER_BY_BEFORE_AT", formattedSQL="ALTER TENANT 'string'::INTERVAL SET CLUSTER SETTING \"position.where.inner.ident\" = \"index_after_order_by_before_at\"": at or near ":": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration368857052
--- FAIL: TestRandomSyntaxGeneration (301.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8299499?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8299499?buildTab=artifacts#/) on master @ [a3d53668ff52c2306f37fcad8e9212ff665ca115](https://github.com/cockroachdb/cockroach/commits/a3d53668ff52c2306f37fcad8e9212ff665ca115):


```
    rsg_test.go:822: 1m20s of 5m0s: 204076 executions, 11444 successful
    rsg_test.go:822: 1m25s of 5m0s: 217449 executions, 12186 successful
    rsg_test.go:822: 1m30s of 5m0s: 230819 executions, 12855 successful
    rsg_test.go:822: 1m35s of 5m0s: 243849 executions, 13499 successful
    rsg_test.go:822: 1m40s of 5m0s: 256475 executions, 14149 successful
    rsg_test.go:822: 1m45s of 5m0s: 269729 executions, 14774 successful
    rsg_test.go:822: 1m50s of 5m0s: 283330 executions, 15434 successful
    rsg_test.go:822: 1m55s of 5m0s: 295954 executions, 16089 successful
    rsg_test.go:822: 2m0s of 5m0s: 308467 executions, 16719 successful
    rsg_test.go:822: 2m5s of 5m0s: 321315 executions, 17319 successful
    rsg_test.go:822: 2m10s of 5m0s: 334006 executions, 17972 successful
    rsg_test.go:822: 2m15s of 5m0s: 347029 executions, 18600 successful
    rsg_test.go:822: 2m20s of 5m0s: 359026 executions, 19214 successful
    rsg_test.go:822: 2m25s of 5m0s: 372775 executions, 19871 successful
    rsg_test.go:822: 2m30s of 5m0s: 386033 executions, 20533 successful
    rsg_test.go:822: 2m35s of 5m0s: 399369 executions, 21208 successful
    rsg_test.go:822: 2m40s of 5m0s: 413099 executions, 21828 successful
    rsg_test.go:822: 2m45s of 5m0s: 425928 executions, 22461 successful
    rsg_test.go:822: 2m50s of 5m0s: 438213 executions, 23037 successful
    rsg_test.go:822: 2m55s of 5m0s: 450187 executions, 23652 successful
    rsg_test.go:822: 3m0s of 5m0s: 462928 executions, 24272 successful
    rsg_test.go:822: 3m5s of 5m0s: 474996 executions, 24860 successful
    rsg_test.go:822: 3m10s of 5m0s: 487763 executions, 25466 successful
    rsg_test.go:822: 3m15s of 5m0s: 500702 executions, 26072 successful
    rsg_test.go:822: 3m20s of 5m0s: 513277 executions, 26719 successful
    rsg_test.go:822: 3m25s of 5m0s: 526919 executions, 27407 successful
    rsg_test.go:822: 3m30s of 5m0s: 539881 executions, 28012 successful
    rsg_test.go:822: 3m35s of 5m0s: 554130 executions, 28723 successful
    rsg_test.go:822: 3m40s of 5m0s: 567668 executions, 29355 successful
    rsg_test.go:822: 3m45s of 5m0s: 581054 executions, 29985 successful
    rsg_test.go:822: 3m50s of 5m0s: 594496 executions, 30627 successful
    rsg_test.go:822: 3m55s of 5m0s: 607947 executions, 31298 successful
    rsg_test.go:822: 4m0s of 5m0s: 621260 executions, 31961 successful
    rsg_test.go:822: 4m5s of 5m0s: 634394 executions, 32646 successful
    rsg_test.go:822: 4m10s of 5m0s: 648549 executions, 33329 successful
    rsg_test.go:822: 4m15s of 5m0s: 661582 executions, 33906 successful
    rsg_test.go:822: 4m20s of 5m0s: 674093 executions, 34533 successful
    rsg_test.go:822: 4m25s of 5m0s: 687122 executions, 35164 successful
    rsg_test.go:822: 4m30s of 5m0s: 700650 executions, 35820 successful
    rsg_test.go:822: 4m35s of 5m0s: 714214 executions, 36467 successful
    rsg_test.go:822: 4m40s of 5m0s: 727893 executions, 37096 successful
    rsg_test.go:822: 4m45s of 5m0s: 741472 executions, 37780 successful
    rsg_test.go:822: 4m50s of 5m0s: 754927 executions, 38427 successful
    rsg_test.go:822: 4m55s of 5m0s: 768043 executions, 39064 successful
    rsg_test.go:822: 5m0s of 5m0s: 780774 executions, 39662 successful
    rsg_test.go:858: 780775 executions, 39662 successful
    rsg_test.go:866: cannot parse output of Format: sql="BACKUP CHAR . * INTO LATEST IN ( 'string' , ident , PLACEHOLDER , PLACEHOLDER , ident ) WITH DETACHED = FALSE", formattedSQL="BACKUP TABLE \"char\".* INTO LATEST IN ('string', 'ident', 'placeholder', 'placeholder', 'ident') WITH ": at or near "EOF": syntax error
    rsg_test.go:305: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration4005415114
--- FAIL: TestRandomSyntaxGeneration (301.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8403717?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/8403717?buildTab=artifacts#/) on master @ [b21379bb56dd206e6f63cc7d07ca72e85db7a4c4](https://github.com/cockroachdb/cockroach/commits/b21379bb56dd206e6f63cc7d07ca72e85db7a4c4):


```
=== RUN   TestRandomSyntaxGeneration
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration3369119495
    test_log_scope.go:79: use -show-logs to present logs inline
    rsg_test.go:822: 5s of 5m0s: 13120 executions, 773 successful
    rsg_test.go:822: 10s of 5m0s: 25887 executions, 1549 successful
    rsg_test.go:822: 15s of 5m0s: 38513 executions, 2329 successful
    rsg_test.go:822: 20s of 5m0s: 52211 executions, 3121 successful
    rsg_test.go:822: 25s of 5m0s: 65619 executions, 3899 successful
    rsg_test.go:822: 30s of 5m0s: 78671 executions, 4648 successful
panic: ReturnType called on TypedExpr with empty typeAnnotation. Was the underlying Expr type-checked before asserting a type of TypedExpr?

goroutine 497572 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.typeAnnotation.assertTyped(...)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/expr.go:152
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Subquery).TypeCheck(0x203006?, {0x0?, 0x0?}, 0xc002427a98?, 0x6d28490?)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/type_check.go:1491 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ArrayFlatten).TypeCheck(0xc0156522a0, {0x6d28538?, 0xc0018a6630?}, 0xc002110b80?, 0x7f85f6ae7ff0?)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/type_check.go:1669 +0x4c
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.TypeCheck({0x6d28538?, 0xc0018a6630?}, {0x6d445e0?, 0xc0156522a0?}, 0x58295ea?, 0xc0136bf200?)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/type_check.go:272 +0x94
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.TypeCheckAndRequire({0x6d28538?, 0xc0018a6630?}, {0x6d445e0?, 0xc0156522a0?}, 0x0?, 0x9aacd60, {0x57d99c3, 0x18})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/type_check.go:282 +0x65
github.com/cockroachdb/cockroach/pkg/sql.(*planner).analyzeExpr(0xc003e74550, {0x6d28538, 0xc0018a6630}, {0x6d445e0?, 0xc0156522a0?}, 0xa18ffe8?, {{0x0, 0x0, 0x0}, {0x0, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/analyze_expr.go:53 +0x133
github.com/cockroachdb/cockroach/pkg/sql.(*planner).planTenantSpec(0x57d99c3?, {0x6d28538?, 0xc0018a6630?}, 0xc002298000?, {0x57d99c3?, 0x740af615b?})
	github.com/cockroachdb/cockroach/pkg/sql/tenant_spec.go:50 +0x254
github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupTenantInfo(0xce4e7ff2f299dd67?, {0x6d28538, 0xc0018a6630}, 0x57d99c3?, {0x57d99c3?, 0xc0157e4948?})
	github.com/cockroachdb/cockroach/pkg/sql/tenant_spec.go:153 +0x2d
github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamingest.alterReplicationJobHook.func1({0x6d28538, 0xc0018a6630}, {0x6d28538?, 0xc00678a660?, 0xf0f000000aa4476?}, 0x6d28538?)
	github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamingest/alter_replication_job.go:153 +0x12d
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:203 +0xa4
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
	github.com/cockroachdb/cockroach/pkg/sql/planhook.go:201 +0x1b2
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #95618 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2]
- #89363 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.2.0]
- #87572 sql/tests: TestRandomSyntaxGeneration failed [DROP OWNED BY timeout] [C-test-failure O-robot T-sql-schema branch-release-22.2]
- #77893 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-sessions branch-release-22.1]
- #74271 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.2]
- #65210 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/tenant
+++ b/pkg/ccl/logictestccl/testdata/logic_test/tenant
@@ -53,3 +53,9 @@ SELECT * FROM crdb_internal.cluster_queries
 
 statement ok
 SELECT * FROM crdb_internal.cluster_contention_events
+
+subtest regression_95612
+
+statement ok
+PREPARE pstmt AS ALTER TENANT 'foo' COMPLETE REPLICATION TO LATEST
+

--- a/pkg/sql/parser/testdata/alter_tenant
+++ b/pkg/sql/parser/testdata/alter_tenant
@@ -190,3 +190,12 @@ ALTER TENANT 'foo' RENAME TO bar
 ALTER TENANT ('foo') RENAME TO (bar) -- fully parenthesized
 ALTER TENANT '_' RENAME TO bar -- literals removed
 ALTER TENANT 'foo' RENAME TO _ -- identifiers removed
+
+# Regression test for #95612
+parse
+ALTER TENANT INTERVAL 'string' MINUTE RESET CLUSTER SETTING ident
+----
+ALTER TENANT ('string'::INTERVAL MINUTE) SET CLUSTER SETTING ident = DEFAULT -- normalized!
+ALTER TENANT ((('string')::INTERVAL MINUTE)) SET CLUSTER SETTING ident = (DEFAULT) -- fully parenthesized
+ALTER TENANT ('_'::INTERVAL MINUTE) SET CLUSTER SETTING ident = DEFAULT -- literals removed
+ALTER TENANT ('string'::INTERVAL MINUTE) SET CLUSTER SETTING ident = DEFAULT -- identifiers removed

--- a/pkg/sql/parser/testdata/backup_restore
+++ b/pkg/sql/parser/testdata/backup_restore
@@ -855,3 +855,12 @@ DETAIL: source SQL:
 RESTORE ROLE foo, bar FROM 'baz'
              ^
 HINT: try \h RESTORE
+
+# Regression test for #95612
+parse
+BACKUP INTO LATEST IN UNLOGGED WITH OPTIONS ( DETACHED = FALSE )
+----
+BACKUP INTO LATEST IN 'unlogged' WITH detached = FALSE -- normalized!
+BACKUP INTO LATEST IN ('unlogged') WITH detached = FALSE -- fully parenthesized
+BACKUP INTO LATEST IN '_' WITH detached = FALSE -- literals removed
+BACKUP INTO LATEST IN 'unlogged' WITH detached = FALSE -- identifiers removed

--- a/pkg/sql/sem/tree/alter_tenant.go
+++ b/pkg/sql/sem/tree/alter_tenant.go
@@ -55,12 +55,67 @@ type TenantSpec struct {
 	All    bool
 }
 
+// alreadyDelimitedAsSyntacticDExpr is an interface that marks
+// Expr types for which there is never an ambiguity when
+// the expression syntax is followed by a non-reserved
+// keyword. When this property is true, that expression
+// can be pretty-printed without enclosing parentheses in
+// a context followed by more non-reserved keywords, and
+// result in syntax that is still unambiguous.
+// That is, given an expression E and an arbitrary following
+// word X, the syntax "E X" is always unambiguously parsed
+// as "(E) X".
+//
+// This property is obviously true of "atomic" expressions such as
+// string and number literals, and also obviously true of
+// well-enclosed expressions "(...)" / "[...]". However, it is not
+// always true of other composite expression types. For example,
+// "A::B" (CastExpr) is not well-delimited because there are
+// identifiers/keywords such that "A::B C" can be parsed as "A::(B
+// C)". Consider "'a'::INTERVAL" and the non-reserved keyword
+// "MINUTE".
+//
+// This property is closely related to the d_expr syntactic rule in
+// the grammar, hence its name. *Approximately* the expression types
+// produced by the d_expr rule tend to exhibit the "well-delimited"
+// property. However, this is not a proper equivalence: certain Expr
+// types are _also_ produced by other parsing rules than d_expr, so
+// inspection of the contents of the Expr object is necessary to
+// determine whether it is well-delimited or not (for example, some
+// FuncExpr objects are well-delimited, and others are not).
+// Therefore, it is not generally correct to assign the property to
+// all the d_expr expression *types*. We can only do so for a few
+// types for which we know that *all possible objects* of that type
+// are well-delimited, such as Subquery, NumVal or Placeholder.
+type alreadyDelimitedAsSyntacticDExpr interface {
+	Expr
+	alreadyDelimitedAsSyntacticDExpr()
+}
+
+func (*UnresolvedName) alreadyDelimitedAsSyntacticDExpr() {}
+func (*ParenExpr) alreadyDelimitedAsSyntacticDExpr()      {}
+func (*Subquery) alreadyDelimitedAsSyntacticDExpr()       {}
+func (*Placeholder) alreadyDelimitedAsSyntacticDExpr()    {}
+func (*NumVal) alreadyDelimitedAsSyntacticDExpr()         {}
+func (*StrVal) alreadyDelimitedAsSyntacticDExpr()         {}
+func (dNull) alreadyDelimitedAsSyntacticDExpr()           {}
+
 // Format implements the NodeFormatter interface.
 func (n *TenantSpec) Format(ctx *FmtCtx) {
 	if n.All {
 		ctx.WriteString("ALL")
 	} else if n.IsName {
+		// Beware to enclose the expression within parentheses if it is
+		// not a simple identifier and is not already enclosed in
+		// parentheses.
+		_, canOmitParentheses := n.Expr.(alreadyDelimitedAsSyntacticDExpr)
+		if !canOmitParentheses {
+			ctx.WriteByte('(')
+		}
 		ctx.FormatNode(n.Expr)
+		if !canOmitParentheses {
+			ctx.WriteByte(')')
+		}
 	} else {
 		ctx.WriteByte('[')
 		ctx.FormatNode(n.Expr)

--- a/pkg/sql/sem/tree/walk.go
+++ b/pkg/sql/sem/tree/walk.go
@@ -994,7 +994,7 @@ func (n *AlterTenantReplication) walkStmt(v Visitor) Statement {
 		}
 		ret.TenantSpec = ts
 	}
-	if n.Cutover != nil {
+	if n.Cutover != nil && n.Cutover.Timestamp != nil {
 		e, changed := WalkExpr(v, n.Cutover.Timestamp)
 		if changed {
 			if ret == n {

